hack/generate-crd.sh

```diff
@@ -0,0 +1,93 @@
+# Read Command Line Parameters
+if [[ $# -ne 2 ]]; then
+    echo "Error: Invalid number of arguments"
+    echo "Usage: ./generate-crd.sh <crd_name> <crd_version>"
+    echo "Example: ./generate-crd.sh newcrd v1"
+    exit 1
+fi
+
+if [[ $GOPATH == "" ]] ; then
+	echo "Error: GOPATH is empty"
+	exit 1
+fi
+
+crd_name=$1
+crd_name_upper="$(tr '[:lower:]' '[:upper:]' <<< ${crd_name:0:1})${crd_name:1}"
+crd_version=$2
+
+# Create directory for CRD definition
+echo "Creating Directories for CRD"
+mkdir "../pkg/api/$crd_name"
+mkdir "../pkg/api/$crd_name/v1"
+
+echo "Copying CRD files from the Sample..."
+cp "../pkg/api/sample/register.txt" "../pkg/api/$crd_name/register.go"
+sed -i "" -e "s/sample/$crd_name/g" "../pkg/api/$crd_name/register.go"
+sed -i "" -e "s/Sample/$crd_name_upper/g" "../pkg/api/$crd_name/register.go"
+echo " > register.go"
+
+cp "../pkg/api/sample/v1/register.txt" "../pkg/api/$crd_name/v1/register.go"
+sed -i "" -e "s/sample/$crd_name/g" "../pkg/api/$crd_name/v1/register.go"
+sed -i "" -e "s/Sample/$crd_name_upper/g" "../pkg/api/$crd_name/v1/register.go"
+echo " > v1/register.go"
+
+cp "../pkg/api/sample/v1/doc.txt" "../pkg/api/$crd_name/v1/doc.go"
+sed -i "" -e "s/sample/$crd_name/g" "../pkg/api/$crd_name/v1/doc.go"
+sed -i "" -e "s/Sample/$crd_name_upper/g" "../pkg/api/$crd_name/v1/doc.go"
+echo " > v1/doc.go"
+
+cp "../pkg/api/sample/v1/types.txt" "../pkg/api/$crd_name/v1/types.go"
+sed -i "" -e "s/sample/$crd_name/g" "../pkg/api/$crd_name/v1/types.go"
+sed -i "" -e "s/Sample/$crd_name_upper/g" "../pkg/api/$crd_name/v1/types.go"
+echo " > v1/types.go"
+
+### THIS CODE IS MODIFIED FROM update-crds.sh ####################
+echo "Cloning Kubernetes Libraries..."
+mkdir -p $GOPATH/src/k8s.io
+pushd $GOPATH/src/k8s.io > /dev/null 2>&1
+  for REPO in code-generator apimachinery api
+  do
+    echo " > git clone git@github.com:kubernetes/${REPO}.git"
+    git clone git@github.com:kubernetes/${REPO}.git > /dev/null 2>&1
+    pushd $REPO > /dev/null 2>&1
+      git pull > /dev/null 2>&1
```

Comment:
Can you please make the similar change that I did in update-crd.sh file?

Reply:
I updated the code